src/data/option/basic.lean

```diff
@@ -105,6 +105,31 @@ by cases x; simp
 
 @[simp] theorem map_id' : option.map (@id α) = id := map_id
 
+/--
+Composing an `option.map` with another `option.map` is equal to
+a single `option.map` of a composition of functions, fully applied.
+This is the reverse direction of `option.comp_map`.
+-/
+@[simp] lemma map_map (h : β → γ) (g : α → β) (x : option α) :
+  option.map h (option.map g x) = option.map (h ∘ g) x :=
+by { cases x; simp only [map_none', map_some'] }
+
+/--
+A single `option.map` of a composition of functions is equal to
+composing an `option.map` with another `option.map`, fully applied.
+This is the reverse direction of `option.map_map`.
+-/
+lemma comp_map (h : β → γ) (g : α → β) (x : option α) :
+  option.map (h ∘ g) x = option.map h (option.map g x) := (map_map _ _ _).symm
```

Comment:
Why is this lemma called `comp_map`, when reading left to right suggests it should be `map_comp`?

Reply:
This is modeling the existing `is_lawful_functor.comp_map`, which is expressed as:
```
(comp_map     : Π {α β γ : Type u} (g : α → β) (h : β → γ) (x : f α), (h ∘ g) <$> x = h <$> g <$> x)
```
I've added `list.comp_map` and `prod.comp_map` in the same statement structure.

Reply:
Ah, so the inconsistency is because one map is infix and the other is a function. I don't really know whether consistency or discoverability is more important here.